logstash-core/src/main/java/org/logstash/config/ir/PipelineConfig.java

```diff
@@ -172,7 +172,6 @@ public void displayDebugInformation() {
             logger.debug("\n\n{}", configPart.getText());
         }
         logger.debug("Merged config");
-        logger.debug("\n\n{}", this.configString());
```

Comment:
review note: this line makes twice same config print (when `--debug` and `--config.debug` enabled) which makes us confusing..., removing.

Reply:
It's duplicated only when the config in composed by just one SourceWithMetadata, but when the config is loaded from multiple files, like when glob operator is used, it logs, every single section, and then also the effective pipeline definition as it's passed down to the compiler.
So I'm little bit doubtful it remove it, especially in cases of strange compilation errors.


Example single file:
```
[2024-04-10T12:22:10,615][DEBUG][org.logstash.config.ir.PipelineConfig] -------- Logstash Config ---------
[2024-04-10T12:22:10,615][DEBUG][org.logstash.config.ir.PipelineConfig] Config from source, source: LogStash::Config::Source::MultiLocal, pipeline_id:: test
[2024-04-10T12:22:10,615][DEBUG][org.logstash.config.ir.PipelineConfig] Config string, protocol: string, id: config_string
[2024-04-10T12:22:10,615][DEBUG][org.logstash.config.ir.PipelineConfig]

input { generator {} } #${I_AM_NOT_HERE}
filter { sleep { time => 100 } }
output {
#${I_AM_NOT_HERE}
  stdout {
    codec => rubydebug
  }
  # ${I_AM_NOT_HERE}
}

[2024-04-10T12:22:10,615][DEBUG][org.logstash.config.ir.PipelineConfig] Merged config
[2024-04-10T12:22:10,615][DEBUG][org.logstash.config.ir.PipelineConfig]

input { generator {} } #${I_AM_NOT_HERE}
filter { sleep { time => 100 } }
output {
#${I_AM_NOT_HERE}
  stdout {
    codec => rubydebug
  }
  # ${I_AM_NOT_HERE}
}

[2024-04-10T12:22:10,616][DEBUG][logstash.agent           ] Converging pipelines state {:actions_count=>1}
```

Example multiple files:
```
[2024-04-10T12:29:45,178][DEBUG][org.logstash.config.ir.PipelineConfig] -------- Logstash Config ---------
[2024-04-10T12:29:45,178][DEBUG][org.logstash.config.ir.PipelineConfig] Config from source, source: LogStash::Config::Source::MultiLocal, pipeline_id:: test
[2024-04-10T12:29:45,178][DEBUG][org.logstash.config.ir.PipelineConfig] Config string, protocol: file, id: /tmp/logstash_1.config
[2024-04-10T12:29:45,179][DEBUG][org.logstash.config.ir.PipelineConfig]

input { generator {} } #${I_AM_NOT_HERE}
filter { sleep { time => 100 } }
[2024-04-10T12:29:45,179][DEBUG][org.logstash.config.ir.PipelineConfig] Config string, protocol: file, id: /tmp/logstash_2.config
[2024-04-10T12:29:45,179][DEBUG][org.logstash.config.ir.PipelineConfig]

output {
     #${I_AM_NOT_HERE}
       stdout {
         codec => rubydebug
       }
       # ${I_AM_NOT_HERE}
}
[2024-04-10T12:29:45,179][DEBUG][org.logstash.config.ir.PipelineConfig] Merged config
[2024-04-10T12:29:45,179][DEBUG][org.logstash.config.ir.PipelineConfig]

input { generator {} } #${I_AM_NOT_HERE}
filter { sleep { time => 100 } }
output {
     #${I_AM_NOT_HERE}
       stdout {
         codec => rubydebug
       }
       # ${I_AM_NOT_HERE}
}
[2024-04-10T12:29:45,179][DEBUG][logstash.agent           ] Converging pipelines state {:actions_count=>1}
```